### PR TITLE
Remove sleep() from hydroponics.dm

### LINKED ISSUE #56435: mutateweed() (and a few other hydroponics procs) contain UNNECESSARY SLEEP()S
sleep() bad

### PR DESCRIPTION
**Pull request by Ghilker**
## About The Pull Request
remove sleep() from a couple of hydroponic procs, use addtimer and callbacks instead, closes #56435

## Why It's Good For The Game
sleeps bad

### CODE REVIEW

File: code/modules/hydroponics/hydroponics.dm [hunk 1]
```diff
@@ -454,13 +452,15 @@
...
-		TRAY_NAME_UPDATE
+		var/message = "<span class='warning'>The mutated weeds in [src] spawn some [myseed.plantname]!</span>"
+		addtimer(CALLBACK(src, .proc/after_wait, message), 0.5 SECONDS)
 	else
 		to_chat(usr, "<span class='warning'>The few weeds in [src] seem to react, but only for a moment...</span>")
 
+/obj/machinery/hydroponics/proc/after_wait(message)
+		update_appearance()
+		visible_message(message)
+		TRAY_NAME_UPDATE
```
Comment (MrMelbert) [inline code/modules/hydroponics/hydroponics.dm:463]: probably needs documentation for this proc and a better name

Comment (Ghilker) [reply code/modules/hydroponics/hydroponics.dm:463]: i'll document it, but i can't think of a good name to give to it

Comment (MrMelbert) [reply code/modules/hydroponics/hydroponics.dm:463]: maybe like "after_mutation"

**Comment by LemonInTheDark** [comment]
Is this all the sleeps in the file?

**Comment by Ghilker** [comment]
> Is this all the sleeps in the file?

yep

**Review by Cyberboss — APPROVED** [review]

**Pull request merged**